### PR TITLE
Move the loadBalancerIP to helm installation as best effort

### PR DESCRIPTION
**Pull request by pooneh-m**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>

> /kind cleanup

**What this PR does / Why we need it**:
Change the `make install` to use allocator loadBalancerIP helm configuration as best effort. If the service is already installed and is just upgrading, then loadBalancerIP is reused and valid allocator certificate gets issued for the service. Otherwise, the current behavior remains.

**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: build/Makefile [hunk 1]
```diff
@@ -312,14 +313,14 @@ install: $(ensure-build-image) install-custom-pull-secret
...
+		--set agones.allocator.http.loadBalancerIP=$(EXTERNAL_IP) \
 		$(HELM_ARGS) \
 		agones $(mount_path)/install/helm/agones/
 
-update-allocation-certs: EXTERNAL_IP ?= $(shell $(DOCKER_RUN) kubectl get services agones-allocator -n agones-system -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
 update-allocation-certs: NAMESPACE ?= default
 update-allocation-certs:
 	-mkdir -p $(agones_path)/build/allocation
-	$(MAKE) install HELM_ARGS="--set agones.allocator.http.loadBalancerIP=$(EXTERNAL_IP)"
+	$(MAKE) install
```
Comment (markmandel) [inline build/Makefile:323]: nit: `install` could be a prerequiste of `update-allocation-certs`.

File: build/Makefile [hunk 2]
```diff
@@ -312,14 +313,14 @@ install: $(ensure-build-image) install-custom-pull-secret
 		--set agones.controller.logLevel=$(LOG_LEVEL) \
 		--set agones.crds.cleanupOnDelete=$(CRD_CLEANUP) \
 		--set agones.featureGates=$(FEATURE_GATES) \
+		--set agones.allocator.http.loadBalancerIP=$(EXTERNAL_IP) \
 		$(HELM_ARGS) \
 		agones $(mount_path)/install/helm/agones/
 
-update-allocation-certs: EXTERNAL_IP ?= $(shell $(DOCKER_RUN) kubectl get services agones-allocator -n agones-system -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
 update-allocation-certs: NAMESPACE ?= default
 update-allocation-certs:
```
Comment (markmandel) [inline build/Makefile:321]:
```suggestion
update-allocation-certs: install
```

Instead of having it inline below.

**Review by markmandel — COMMENTED** [review]
Small nit, but otherwise, looks like an excellent change!

File: build/Makefile (hunk 2 above)
Comment (pooneh-m) [reply build/Makefile:322]: Thanks for the suggestion. Applied the change.

**Review by markmandel — APPROVED** [review]
Nice!

**Pull request merged**